wandb/sdk/wandb_metadata.py

```diff
@@ -0,0 +1,563 @@
+from __future__ import annotations
+
+import sys
+from contextlib import contextmanager
+from datetime import datetime, timezone
+from typing import Any, Callable
+
+from google.protobuf.timestamp_pb2 import Timestamp
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+
+from wandb.proto import wandb_internal_pb2
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+class DiskInfo(BaseModel, validate_assignment=True):
+    total: int | None = None
+    used: int | None = None
+
+    def to_proto(self) -> wandb_internal_pb2.DiskInfo:
+        return wandb_internal_pb2.DiskInfo(
+            total=self.total if self.total is not None else 0,
+            used=self.used if self.used is not None else 0,
```

Comment:
```suggestion
            total=self.total or else 0,
            used=self.used or else 0,
```
should also work in this case because use/total can only be int and None